src/cogs/jaegeraccounts.py

```diff
@@ -162,23 +160,18 @@ async def book(self, ctx):
 
         sheet_data = await SheetData.from_url(self.bot, ctx, url)
 
+        # Necessary because author.nick is None if the user has not changed his name on the server      
+        name = ctx.author.name
+        if ctx.author.nick is not None:
+            name = ctx.author.nick
+
         # Try to assign accounts to the person that last had it as often as possible.
         for account in sheet_data.accounts:
-            # Necessary because author.nick is None if the user has not changed his name on the server
-            if ctx.author.nick is not None:
-                name = ctx.author.nick
-            else:
-                name = ctx.author.name
-
-            if account.last_user == name:
-                # It is more efficient to repeat a good part of the user_has_account function than to call it
-                if account.is_booked:
-                    await ctx.reply(f"You have already been assigned: `{account.name}`.\n"
+            if account.last_user == name and account.is_booked:
+                await ctx.reply(f"You have already been assigned: `{account.name}`.\n"
                                     "Please check your PMs for the login details.")
-                    return
-                else:
-                    await ctx.author.send(embed=account.embed)
-                    return
+                return
+
```

Comment:
```suggestion
            else:
                await ctx.author.send(embed=account.embed)
                return
```
This should fix things